sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/ViewRegistry.java

```diff
@@ -8,59 +8,38 @@
 import io.opentelemetry.sdk.metrics.aggregator.AggregatorFactory;
 import io.opentelemetry.sdk.metrics.common.InstrumentDescriptor;
 import io.opentelemetry.sdk.metrics.common.InstrumentType;
-import io.opentelemetry.sdk.metrics.view.InstrumentSelector;
 import io.opentelemetry.sdk.metrics.view.View;
 import java.util.EnumMap;
 import java.util.LinkedHashMap;
 import java.util.Map;
-import java.util.concurrent.locks.ReentrantLock;
 import java.util.regex.Pattern;
+import javax.annotation.concurrent.Immutable;
 
 /**
- * Central location for Views to be registered. Registration of a view should eventually be done via
- * the {@link SdkMeterProvider}.
- *
- * <p>This class uses copy-on-write for the registered views to ensure that reading threads get
- * never blocked.
+ * Central location for Views to be registered. Registration of a view is done via the {@link
+ * SdkMeterProviderBuilder}.
  */
+@Immutable
 final class ViewRegistry {
-  private static final LinkedHashMap<Pattern, View> EMPTY_CONFIG = new LinkedHashMap<>();
   static final View CUMULATIVE_SUM =
       View.builder().setAggregatorFactory(AggregatorFactory.sum(true)).build();
   static final View SUMMARY =
       View.builder().setAggregatorFactory(AggregatorFactory.minMaxSumCount()).build();
   static final View LAST_VALUE =
       View.builder().setAggregatorFactory(AggregatorFactory.lastValue()).build();
 
-  // The lock is used to ensure only one updated to the configuration happens at any moment.
-  private final ReentrantLock lock = new ReentrantLock();
-  private volatile EnumMap<InstrumentType, LinkedHashMap<Pattern, View>> configuration;
+  private final EnumMap<InstrumentType, LinkedHashMap<Pattern, View>> configuration;
 
-  ViewRegistry() {
+  ViewRegistry(EnumMap<InstrumentType, LinkedHashMap<Pattern, View>> configuration) {
     this.configuration = new EnumMap<>(InstrumentType.class);
-    configuration.put(InstrumentType.COUNTER, EMPTY_CONFIG);
-    configuration.put(InstrumentType.UP_DOWN_COUNTER, EMPTY_CONFIG);
-    configuration.put(InstrumentType.VALUE_RECORDER, EMPTY_CONFIG);
-    configuration.put(InstrumentType.SUM_OBSERVER, EMPTY_CONFIG);
-    configuration.put(InstrumentType.UP_DOWN_SUM_OBSERVER, EMPTY_CONFIG);
-    configuration.put(InstrumentType.VALUE_OBSERVER, EMPTY_CONFIG);
+    // make a copy for safety
```

Comment:
I would generally not do this for private constructors, but no big deal either way

Reply:
I probably wouldn't normally, either, but since it's a one-time configuration cost to make the defensive copy, this will protect us a bit if somehow this class ends up more public in the future.